### DISCUSSION TITLE
TCountView missing

### DISCUSSION THREAD

**Discussion by paule32** [magiblot/tvision#146] [Q&A] [answered]
Hello,
I using TDialog for custom TWindow view.
The TDialog don't insert a window counter, so I would like to do this by my own, but I fail.
In the Pascal Dokumentation of Turbo Vision, there is a TCountView.
So, the Question is, how to use this class, if available ?
Here is a little bit of Code:
```
TCountView    * counterView;
...
insert(counterView = new TCountView( TRect( 5,0, 12,1) ));
counterView->SetCount(counter);

```

**Comment by magiblot** [comment, marked as answer]
Hi Jens!

To use the built-in window counters, you just have to initialize the `number` field in your dialog (inherited from `TWindow`). The code for drawing the window counter is in `TFrame`:

https://github.com/magiblot/tvision/blob/0917f04206fcfa1973dc49547af74e7399241899/source/tvision/tframe.cpp#L69-L79

However, as you can see, it only displays window numbers smaller than 10.

To calculate the window number, I guess the idiomatic way would be to define a custom broadcast event (e.g. `cmGetMaxWindowNum`) which is only handled by your custom dialogs. For example, it could be something like this:

```c++
void TMyApp::createNewWindow() {
    int maxWindowNum = (int)(size_t) message(this, evBroadcast, cmGetMaxWindowNum, 0);
    int newWindowNum = maxWindowNum + 1;
    auto *myWindow = new TMyWindow(TRect(/* ... */), "Title", newWindowNum);
    insert(myWindow);
}

void TMyWindow::handleEvent(TEvent &ev) {
    TWindow::handleEvent(ev); // Or whichever other class you are inheriting from.
    if (ev.what == evBroadcast && ev.message.command == cmGetMaxWindowNum) {
        if (number > ev.message.infoInt) // 'number' field is inherited from TWindow.
            ev.message.infoInt = number;
        // Do not clearEvent(), since this is a broadcast event which other windows should also handle.
    }
}
```
